spark/src/main/scala/org/apache/spark/shuffle/sort/RowPartition.scala

```diff
@@ -36,7 +36,10 @@ class RowPartition(initialSize: Int) {
   def getRowSizes: Array[Int] = rowSizes.toArray
 
   def reset(): Unit = {
-    rowAddresses.clear()
-    rowSizes.clear()
+    rowAddresses = null
```

Comment:
nit: why do we need to set this to null first?

Reply:
Oh, it is not needed. Let me remove it.